### PR TITLE
ledger-autosync: init at 0.2.3

### PR DESCRIPTION
**Pull request by nlewo**

### CODE REVIEW

**Comment by FRidh** [comment]
hi @nlewo , could you move this package to `python-packages.nix`. You can still call it from `top-level.nix`.

File: pkgs/applications/office/ledger-autosync/default.nix [hunk 1]
```diff
@@ -0,0 +1,27 @@
+{ stdenv, fetchurl, pythonPackages, python, ledger, which } :
+
+pythonPackages.buildPythonPackage rec {
+  name = "ledger-autosync";
+  version = "0.2.3";
+  src = fetchurl {
+    url = "https://pypi.python.org/packages/source/l/ledger-autosync/ledger-autosync-${version}.tar.gz";
+    md5 = "070930f80b24cb5a4861952c62d9f100";
```
Comment (FRidh) [inline pkgs/applications/office/ledger-autosync/default.nix]: please use a sha256 here

**Comment by nlewo** [comment]
Hi @FRidh , I thought python-packages.nix was only used for python libraries (and not for python programs). Humm, it's hard to know where to put nix expressions...

Could you please confirm we also have to put python programs into python-packages.nix?

**Comment by FRidh** [comment]
There are actually many opinions on that topic Several days ago there was for instance again a discussion on the mailing list about flattening the repository and doing with less grouping.

[I would prefer](https://github.com/NixOS/nixpkgs/issues/11567) to have all Python packages, and at the very least all those available via PyPI, in `python-packages.nix` because it's more convenient from a maintenance point of view, because it allows you more easily to use a different Python version of the package, and because it reduces the chance on certain errors when used in combination with other Python packages. And hopefully someday we have a fully functioning `pypi2nix`. It would then be a waste if you still had to update packages manually because they were put in a different part of the repository :)

**Comment by nlewo** [comment]
Thanks @FRidh for your explanations. I moved this package to python-packages.nix as you recommended.

If your point of view is also "the nix community" point of view, it would be pertinent to update nix python package documentation to mention the fact that python packages MUST be implemented in python-packages.nix. Currently, the documentation says "Most of Python packages that use buildPythonPackage are defined in pkgs/top-level/python-packages.nix" and the python contributing guidelines doesn't mention the location of python packages.

**Comment by FRidh** [comment]
@nlewo I agree some clarity on the topic in the manual is needed.

By the way, regarding the tests. `hledger` seems to be available, so you could add it as buildInput in order to make the tests run.

**Comment by nlewo** [comment]
Thanks @FRidh  for the merge. Regarding test dependencies, the main problem is the python ledger module which is a mess to build (https://groups.google.com/forum/#!msg/ledger-cli/bcfZ7Rf68bY/TtcjZgPlqekJ) and not really useful I guess.

**Pull request merged**